src/environmentd/tests/server.rs

```diff
@@ -957,3 +958,80 @@ fn test_mz_system_user_admin() {
             .get::<_, String>(0)
     );
 }
+
+#[test]
+#[cfg_attr(miri, ignore)] // too slow
+fn test_ws_passes_options() {
+    let server = util::start_server(util::Config::default()).unwrap();
+
+    // Create our WebSocket.
+    let ws_url = Url::parse(&format!(
+        "ws://{}/api/experimental/sql",
+        server.inner.http_local_addr()
+    ))
+    .unwrap();
+    let (mut ws, _resp) = tungstenite::connect(ws_url).unwrap();
+    let options = BTreeMap::from([(
+        "application_name".to_string(),
+        "billion_dollar_idea".to_string(),
+    )]);
+    util::auth_with_ws(&mut ws, Some(options));
+
+    // Query to make sure we get back the correct session var, which should be
+    // set from the options map we passed with the auth.
+    let json = "{\"query\":\"SHOW application_name;\"}";
```

Comment:
This test LGTM, but at also reveals maybe a problem in our WS protocol. The pgwire protocol, on startup, responds with all of the session settings. We don't do that here, forcing a SHOW instead to display one. I'm not sure which is better (to auto show on startup or not), but it's a notable difference.

Reply:
I feel like we should probably try to follow the pgwire protocol? I made a task to track this: https://github.com/MaterializeInc/materialize/issues/18503